src/navigation/menu_bar.py

```diff
@@ -468,20 +469,57 @@ async def coroutine(self: MenuNav) -> None:
         ensure_future(coroutine(self))
 
     def do_exit(self) -> None:
-        """Exit"""
-        settings_path = os.path.join(NOTES_DIR, ".user_setting.json")
-        with open(settings_path, "r") as f:
-            user = json.load(f)
+        """Exit app, with warning if current file unsaved"""
 
-        self.application_state.user_settings[
-            "last_path"
-        ] = self.application_state.current_path
+        async def coroutine(self: MenuNav) -> None:
+            title = "Unsaved Changes"
+            # If file previously saved, check if current version matches saved
+            if (
+                current_path_valid := self.application_state.current_path
+                and os.path.exists(self.application_state.current_path)
+            ):
+                text = "\n".join(
+                    (
+                        f"The file {self.application_state.current_path}",
+                        "contains unsaved changes. Save before exit?",
+                    )
+                )
+                with open(self.application_state.current_path) as f:
+                    written = f.read()
+                unsaved_changes = written != self.text_field.text
+            # If file not previously saved, warn if contains any text
+            else:
+                text = "This file has not yet been saved. Save before exit?"
+                unsaved_changes = self.text_field.text != ""
+            if unsaved_changes:
+                dialog = SaveExitDialog(title=title, text=text)
+                choice = await self.show_dialog_as_float(dialog)
+            else:
+                choice = "nosave"
+            if choice != "cancel":
+                if choice == "save":
+                    # If not yet saved, generate generic name to save to
+                    if not current_path_valid:
+                        self.application_state.current_path = os.path.join(
+                            NOTES_DIR, get_unique_filename(NOTES_DIR) + ".txt"
```

Comment:
```suggestion
                            NOTES_DIR, get_unique_filename(NOTES_DIR)
```
You need to remove the extension. `get_unique_filename` looks to see if `Thought Box Note` is in the directory, then you are writing to `Thought Box Note.txt`, which is a different name. So the next time it is called, it suggests `Thought Box Note` again, and you end up overwriting `Thought Box Note.txt`.